pkg/rtc/participant.go

```diff
@@ -1035,9 +1037,15 @@ func (p *ParticipantImpl) setupTransportManager() error {
 	}
 
 	tm.OnICEConfigChanged(func(iceConfig types.IceConfig) {
-		p.lock.RLock()
+		p.lock.Lock()
 		onICEConfigChanged := p.onICEConfigChanged
-		p.lock.RUnlock()
+		if iceConfig.PreferSub == types.PreferTls {
```

Comment:
should we have an else block that unsets this when PreferTls is disabled?

Reply:
Will add it just.